helfomat-infrastructure-typo3/src/main/java/de/helfenkannjeder/helfomat/infrastructure/typo3/LegacyDataSourceConfig.java

```diff
@@ -0,0 +1,70 @@
+package de.helfenkannjeder.helfomat.infrastructure.typo3;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.orm.jpa.hibernate.SpringImplicitNamingStrategy;
+import org.springframework.boot.orm.jpa.hibernate.SpringPhysicalNamingStrategy;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+import org.springframework.transaction.PlatformTransactionManager;
+
+import javax.sql.DataSource;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Valentin Zickner
+ */
+@Configuration
+@EnableJpaRepositories(
+    basePackageClasses = Typo3OrganisationRepository.class,
+    entityManagerFactoryRef = "legacyEntityManager",
+    transactionManagerRef = "legacyTransactionManager"
+)
+public class LegacyDataSourceConfig {
+
+    @Bean
+    @Primary
+    public LocalContainerEntityManagerFactoryBean legacyEntityManager(@Qualifier("legacyDataSource") DataSource legacyDataSource) {
+        LocalContainerEntityManagerFactoryBean entityManagerFactoryBean = new LocalContainerEntityManagerFactoryBean();
+        entityManagerFactoryBean.setDataSource(legacyDataSource);
+        entityManagerFactoryBean.setPackagesToScan(this.getClass().getPackage().getName());
+
+        HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+        entityManagerFactoryBean.setJpaVendorAdapter(vendorAdapter);
+        Map<String, String> jpaProperties = new HashMap<>();
+        jpaProperties.put("hibernate.hbm2ddl.auto", "none");
+        jpaProperties.put("hibernate.physical_naming_strategy", SpringPhysicalNamingStrategy.class.getName());
+        jpaProperties.put("hibernate.implicit_naming_strategy", SpringImplicitNamingStrategy.class.getName());
+        entityManagerFactoryBean.setJpaPropertyMap(jpaProperties);
+
+        return entityManagerFactoryBean;
+    }
+
+    @Bean
+    public PlatformTransactionManager legacyTransactionManager(@Qualifier("legacyEntityManager") LocalContainerEntityManagerFactoryBean legacyEntityManager) {
```

Comment:
Do you really have to write such stuff into the code? Is the framework that dumb? :D

Reply:
While using two data sources, unfortunately yes :(